File: core/models/room.py
from django.db import models


class Room(models.Model):
    floor = models.ForeignKey('Floor', on_delete=models.CASCADE, related_name='rooms')

    name = models.CharField(max_length=32)
    description = models.TextField(blank=True, default='')
    tags = models.JSONField(default=list)

    width = models.FloatField()
    height = models.FloatField()
    pos_x = models.FloatField()
    pos_y = models.FloatField()
    orientation = models.FloatField()

    def __str__(self):
        return f'{self.name} ({self.floor.name})'

File: app/src/models/room.ts
export interface Room {
    id: number;
    name: string;
    description: string;
    tags: string[];

    width: number;
    height: number;
    pos_x: number;
    pos_y: number;
    orientation: number;

    floor: number;
}

export interface DisplayRoom extends Room {
    is_blocked: boolean;
    is_blocked_from: string;
    is_blocked_until: string;
    booking_user?: string;
}

export function isDisplayRoom(obj: any): obj is DisplayRoom {
    return obj != null && obj.is_blocked != null;
}
